pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -101,7 +101,17 @@ func (r *SubnetPortReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		r.StatusUpdater.IncreaseUpdateTotal()
 
 		old_status := subnetPort.Status.DeepCopy()
-		isExisting, isParentResourceTerminating, nsxSubnetPath, err := r.CheckAndGetSubnetPathForSubnetPort(ctx, subnetPort)
+		subnetCR, isParentResourceTerminating, err := r.getSubnetCR(ctx, subnetPort)
+		if isParentResourceTerminating {
+			err = errors.New("parent resource is terminating, SubnetPort cannot be created")
+			r.StatusUpdater.UpdateFail(ctx, subnetPort, err, "", setSubnetPortReadyStatusFalse, r.SubnetPortService, r.restoreMode)
+			return common.ResultNormal, err
+		}
+		if err != nil {
+			r.StatusUpdater.UpdateFail(ctx, subnetPort, err, "Failed to get Subnet CR", setSubnetPortReadyStatusFalse, r.SubnetPortService, r.restoreMode)
+			return common.ResultRequeue, err
```

Comment:
if SubnetPort created on a not-found Subnet CR,  seems the SubnetPort will be requeue forever.

Reply:
As discussed in chat, we requeue the SubnetPort with not-found Subnet CR in 9.0 which supports SubnetPort CR created before Subnet CR. If we do not requeue it any more, it will be inconsistent with 9.0.
We prefer not to change this now, but a follow up PR can consider to 
- not requeue the SubnetPort with not-found Subnet CR, but add a SubnetPort webhook to prevent user creating a SubnetPort with non-existed Subnet CR
- not requeue the SubnetPort with not-found Subnet CR, and add a Subnet handler to enqueue the corresponding SubnetPort  upon Subnet creation
@lxiaopei Please let me know if you have preference on the 2 solution, thanks!

Reply:
I prefer option 1 to let user know when creating SubnetPort that the specified Subnet not exist.
I think most users do not want to create SubnetPort CR using a non-exist Subnet.